### PR TITLE
Use real UUID for obfuscating username

### PR DESCRIPTION
**Pull request by yli-sugarcrm**
Closes #215

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -815,6 +815,14 @@ describe('Thorn', () => {
             });
 
             it('should use internal usernames for server interactions', function*() {
+                let uuidv4Format = [
+                    '[a-fA-F0-9]{8}',
+                    '[a-fA-F0-9]{4}',
+                    '4{1}[a-fA-F0-9]{3}',
+                    '[89abAB]{1}[a-fA-F0-9]{3}',
+                    '[a-fA-F0-9]{12}'
```
Comment (rhoggSugarcrm) [inline tests/index.js]: Please place a trailing comma at the end of this line.

For future reference, you can run `gulp lint` before submitting a PR to make sure your code is style-guide compliant.

File: tests/index.js [hunk 2]
```diff
@@ -815,6 +815,14 @@ describe('Thorn', () => {
             });
 
             it('should use internal usernames for server interactions', function*() {
+                let uuidv4Format = [
+                    '[a-fA-F0-9]{8}',
+                    '[a-fA-F0-9]{4}',
+                    '4{1}[a-fA-F0-9]{3}',
```
Comment (rhoggSugarcrm) [inline tests/index.js]: We can delete any `{1}`'s as they're redundant.

File: tests/index.js [hunk 3]
```diff
@@ -815,6 +815,14 @@ describe('Thorn', () => {
...
+                    '[89abAB]{1}[a-fA-F0-9]{3}',
```
Comment (rhoggSugarcrm) [inline tests/index.js]: For clarification: the 89AB is the `variant` part of the RFC 4122 UUID. Per [the spec](http://www.ietf.org/rfc/rfc4122.txt):

```
Msb0  Msb1  Msb2  Description
    0     x     x    Reserved, NCS backward compatibility.
    1     0     x    The variant specified in this document.
    1     1     0    Reserved, Microsoft Corporation backward compatibility
    1     1     1    Reserved for future definition.
```

8, 9, A, and B are the only hexadecimal digits that start with `10` in binary; anything higher than that starts with `11` and anything lower than that starts with `01`.

**Comment by jcsmorais** [comment]
Username have a maximum length of 60 chars, if we're now using UUID's (which take more than half-of that, 36 chars) should we add something to the documentation to explicitly say these should be under 24 chars?

**Pull request merged**